src/acp/acp.ts

```diff
@@ -0,0 +1,233 @@
+ * @returns A file and the ACR that applies to it, if available to the authenticated user, and the APRs that are referred to therein, if available to the authenticated user.
+ */
+export async function getFileWithAcp(
+  url: Url | UrlString,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): Promise<File & WithAcp> {
+  const urlString = internal_toIriString(url);
+  const config = {
+    ...internal_defaultFetchOptions,
+    ...options,
+  };
+
+  const file = await getFile(urlString, config);
+  const acp = await fetchAcp(file, config);
+  return Object.assign(file, acp);
+}
+
+/**
+ * ```{note} The Web Access Control specification is not yet finalised. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Retrieve information about a Resource, its associated Access Control Resource (if available to
+ * the current user), and all the Access Control Policies referred to therein, if available to the
+ * current user, without fetching the Resource itself.
+ *
+ * @param url URL of the Resource about which to fetch its information.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns Metadata describing a Resource, and the ACR that applies to it, if available to the authenticated user, and the APRs that are referred to therein, if available to the authenticated user.
+ */
+export async function getResourceInfoWithAcp(
+  url: Url | UrlString,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): Promise<WithResourceInfo & WithAcp> {
+  const urlString = internal_toIriString(url);
+  const config = {
+    ...internal_defaultFetchOptions,
+    ...options,
+  };
+
+  const resourceInfo = await getResourceInfo(urlString, config);
+  const acp = await fetchAcp(resourceInfo, config);
+  return Object.assign(resourceInfo, acp);
+}
+
+export type WithAcp = {
+  internal_acp:
+    | {
+        acr: AccessControlResource;
+        aprs: Record<UrlString, PolicyDataset | null>;
+      }
+    | {
+        acr: null;
+      };
+};
+export type WithAccessibleAcr = WithAcp & {
+  internal_acp: {
+    acr: Exclude<WithAcp["internal_acp"]["acr"], null>;
+  };
+};
+
+async function fetchAcp(
+  resource: WithResourceInfo,
+  options: Partial<typeof internal_defaultFetchOptions>
+): Promise<WithAcp> {
+  if (!hasLinkedAcr(resource)) {
+    return {
+      internal_acp: {
+        acr: null,
+      },
+    };
+  }
+  let acr: SolidDataset;
+  try {
+    acr = await getSolidDataset(
+      resource.internal_resourceInfo.linkedResources[acp.accessControl][0],
```

Comment:
I assume the spec mandates a resource to only have one ACR, and if so maybe a comment here could explain the `[0]`which may seem arbitrary otherwise.

Reply:
Good point: 68ea396cfc4967d2cccd89c01c6048eb5a62ffc8